File: app/src/test/java/coolpharaoh/tee/speicher/tea/timer/views/new_tea/InputValidatorTest.java
package coolpharaoh.tee.speicher.tea.timer.views.new_tea;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.ArgumentMatchers.anyInt;
import static org.mockito.Mockito.when;

import android.app.Application;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnitRunner;

@RunWith(MockitoJUnitRunner.class)
public class InputValidatorTest {

    @Mock
    private Application application;

    private InputValidator inputValidator;

    @Before
    public void setUp() {
        when(application.getString(anyInt())).thenReturn("ErrorMessage");
        inputValidator = new InputValidator(application, System.out::println);
    }

    @Test
    public void nameIsNotEmptyReturnsFalse() {
        assertThat(inputValidator.nameIsNotEmpty("")).isFalse();
    }

    @Test
    public void nameIsNotEmptyReturnsTrue() {
        assertThat(inputValidator.nameIsNotEmpty("Tea")).isTrue();
    }

    @Test
    public void nameIsValidReturnsFalse() {
        final char[] data = new char[350];
        final String largeName = new String(data);
        assertThat(inputValidator.nameIsValid(largeName)).isFalse();
    }

    @Test
    public void nameIsValidReturnsTrue() {
        assertThat(inputValidator.nameIsValid("Tea")).isTrue();
    }

    @Test
    public void varietyIsValidReturnsFalse() {
        final char[] data = new char[50];
        final String largeVariety = new String(data);
        assertThat(inputValidator.varietyIsValid(largeVariety)).isFalse();
    }

    @Test
    public void varietyIsValidReturnsTrue() {
        assertThat(inputValidator.varietyIsValid("Variety")).isTrue();
    }
}

File: app/src/test/java/coolpharaoh/tee/speicher/tea/timer/views/new_tea/InputValidatorTest.kt
package coolpharaoh.tee.speicher.tea.timer.views.new_tea

import android.app.Application
import coolpharaoh.tee.speicher.tea.timer.core.print.Printer
import io.mockk.every
import io.mockk.impl.annotations.MockK
import io.mockk.junit5.MockKExtension
import org.assertj.core.api.Assertions.*
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith

@ExtendWith(MockKExtension::class)
internal class InputValidatorTest {
    @MockK
    lateinit var application: Application

    private var inputValidator: InputValidator? = null

    @BeforeEach
    fun setUp() {
        inputValidator = InputValidator(application, object : Printer { override fun print(message: String?) { println(message) } })
    }

    @Test
    fun nameIsNotEmptyReturnsFalse() {
         every { application.getString(any()) } returns "ErrorMessage"

        assertThat(inputValidator!!.nameIsNotEmpty("")).isFalse
    }

    @Test
    fun nameIsNotEmptyReturnsTrue() {
        assertThat(inputValidator!!.nameIsNotEmpty("Tea")).isTrue
    }

    @Test
    fun nameIsValidReturnsFalse() {
         every { application.getString(any()) } returns "ErrorMessage"

        val data = CharArray(350)
        val largeName = String(data)
        assertThat(inputValidator!!.nameIsValid(largeName)).isFalse
    }

    @Test
    fun nameIsValidReturnsTrue() {
        assertThat(inputValidator!!.nameIsValid("Tea")).isTrue
    }

    @Test
    fun varietyIsValidReturnsFalse() {
         every { application.getString(any()) } returns "ErrorMessage"

        val data = CharArray(50)
        val largeVariety = String(data)
        assertThat(inputValidator!!.varietyIsValid(largeVariety)).isFalse
    }

    @Test
    fun varietyIsValidReturnsTrue() {
        assertThat(inputValidator!!.varietyIsValid("Variety")).isTrue
    }
}